app/models/creator.rb

```diff
@@ -1,2 +1,67 @@
 class Creator < ActiveRecord::Base
+  validates_presence_of :name, :uid, :provider
+  validates_uniqueness_of :uid
+  has_many :categories
+  has_many :users, through: :categories
+  has_many :videos
+  has_many :tweets
+
+  def self.find_or_create(params)
+    creator = self.find_by(uid: params["uid"], provider: params["provider"])
+    if !creator.nil?
+      return creator
+    else
+      Creator.transaction do
+        creator = Creator.new(
+            name: params["name"],
+            provider: params["provider"],
+            uid: params["uid"]
+              )
+          if creator.provider == "vimeo"
+            if !params["profile_pic"].nil?
+              creator.profile_pic = params["profile_pic"]["sizes"][2]["link"]
+            else
+              creator.profile_pic = "vimeo_default_image.png"
+            end
+            creator.save
+            video_data_array = creator.get_videos_info
+            Video.create_videos_for_creator_from_hashes(video_data_array, creator)
+          elsif creator.provider == "twitter"
+            if !params["profile_pic"].nil?
+              creator.profile_pic = params["profile_pic"]
+            else
+              creator.profile_pic = "twitter_default_image.png"
+            end
+            creator.save
+            Tweet.find_or_create_tweets(creator)
+          end
+        end
+    end
+    if creator.save
+      return creator
+    else
+      return nil
+    end
+  end
```

Comment:
This method is really large. I find it helpful to ask myself "is this method in doing more than one thing?". It's nice to break up methods by responsibility, doing so reduces over-dependency and increases readability for those inheriting your code. 
You'll find in the "real world that you're rarely writing code in a vacuum -- everyone is inheriting on another's code and favoring clarity and separation of responsibility is very helpful for others.
Example:
You can extract your "vimeo" and "twitter" logic into other methods that are then called from this `find_or_create` method